File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/interior-detail.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyInteriorDetailTypeOfPlumbings } from "./interior-detail/InspectionOrderPropertyInteriorDetailTypeOfPlumbings";
import { InspectionOrderPropertyInteriorDetailQualityClassUpgrades } from "./interior-detail/InspectionOrderPropertyInteriorDetailQualityClassUpgrades";
import { InspectionOrderPropertyInteriorDetailKitchenBathCounters } from "./interior-detail/InspectionOrderPropertyInteriorDetailKitchenBathCounters";
import { InspectionOrderPropertyInteriorDetailKitchenBathCabinets } from "./interior-detail/InspectionOrderPropertyInteriorDetailKitchenBathCabinets";
import { InspectionOrderPropertyInteriorDetailFlooringTypes } from "./interior-detail/InspectionOrderPropertyInteriorDetailFlooringTypes";

export class InspectionOrderPropertyInteriorDetail {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    id?: string;
    flooringTypes?: InspectionOrderPropertyInteriorDetailFlooringTypes[];
    kitchenBathCabinets?: InspectionOrderPropertyInteriorDetailKitchenBathCabinets[];
    kitchenBathCounters?: InspectionOrderPropertyInteriorDetailKitchenBathCounters[];
    qualityClassUpgrades?: InspectionOrderPropertyInteriorDetailQualityClassUpgrades[];
    interiorDetailComment?: string;
    typeOfPlumbings?: InspectionOrderPropertyInteriorDetailTypeOfPlumbings[];
    electricalServiceAmperage?: string;
    waterHeaterLocation?: string;
    laundryLocation?: string;

    public constructor(init?:Partial<InspectionOrderPropertyInteriorDetail>) {
        Object.assign(this, init);
    }
}

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/InteriorDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyInteriorDetail
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }

        public ICollection<InspectionOrderPropertyInteriorDetailFlooringTypes> FlooringTypes { get; set; }
        public ICollection<InspectionOrderPropertyInteriorDetailKitchenBathCabinets> KitchenBathCabinets { get; set; }
        public ICollection<InspectionOrderPropertyInteriorDetailKitchenBathCounters> KitchenBathCounters { get; set; }
        public ICollection<InspectionOrderPropertyInteriorDetailQualityClassUpgrades> QualityClassUpgrades { get; set; }
        public string InteriorDetailComment { get; set; }
        public ICollection<InspectionOrderPropertyInteriorDetailTypeOfPlumbings> TypeOfPlumbings { get; set; }
        public string ElectricalServiceAmperage { get; set; }
        public string WaterHeaterLocation { get; set; }
        public string LaundryLocation { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.InteriorDetail).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyInteriorDetail>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{nameof(InspectionOrderPropertyInteriorDetail)}")
                .OnDelete(DeleteBehavior.Cascade);

            InspectionOrderPropertyInteriorDetailFlooringTypes.BuildModel(modelBuilder);

            InspectionOrderPropertyInteriorDetailKitchenBathCabinets.BuildModel(modelBuilder);

            InspectionOrderPropertyInteriorDetailKitchenBathCounters.BuildModel(modelBuilder);

            InspectionOrderPropertyInteriorDetailQualityClassUpgrades.BuildModel(modelBuilder);

            InspectionOrderPropertyInteriorDetailTypeOfPlumbings.BuildModel(modelBuilder);
        }
    }
}